game/hud/src/widgets/Crafting/components/Tools/index.tsx

```diff
@@ -35,7 +35,7 @@ const select = (state: GlobalState, props: ToolsProps) : ToolsPropsRedux => {
 
 export interface ToolsProps extends ToolsPropsRedux {
```

Comment:
Currently, the project is not building because of these typings. To fix the build issue, you need to explicitly declare Redux props and the component's own props. Then combine them in a Type.

Something like 
`
export interface ToolsPropsRedux {
  countdown: number;
}

export interface ToolsOwnProps {
  style?: Partial<ToolsStyles>;
  dispatch: (action: any) => void;
  refresh: () => void;
}

const select = (state: GlobalState, ownProps: ToolsOwnProps): ToolsPropsRedux => {
  return { countdown: state.ui.countdown };
};

export type ToolsProps = ToolsPropsRedux & ToolsOwnProps;
`

And then use the ToolsProps type for the component. This should fix the build issues.

Also using this method, you shouldn't make the redux props/dispatch props optional.

Reply:
Humm... tried that, and I get a similar build error 

```src/widgets/Crafting/components/Tools/index.tsx(226,32): error TS2345: Argument of type 'typeof Tools' is not assignable to parameter of type 'Component<ToolsOwnProps & DispatchProp<any> & ToolsPropsRedux>'.```

It's complaining about Tools in

```
export default connect(select)(Tools);
```

Reply:
Never mind!   Had not removed the `?` on `countdown: number;` now that I have, it compiles.